js/src/forum/components/DiscussionList.js

```diff
@@ -187,8 +201,22 @@ export default class DiscussionList extends Component {
     this.loading = false;
     this.moreResults = !!results.payload.links.next;
 
+    // Construct a URL to this discussion with the updated page, then
+    // replace it into the window's history and our own history stack.
     m.lazyRedraw();
 
+    const query = m.route.parseQueryString(document.location.search);
+
+    if (this.page !== query.page) {
+      if (this.page !== 1) query.page = this.page;
```

Comment:
Couldn't we set this at the `this.page` definition already?

Reply:
@luceos What do you mean ? `this.page` keeps track of current page, and this line adds it to the query string if it isn't `1`. Do you mean have `this.page` set to undefined (i.e. unset) ?

Reply:
I meant, can we use the query page earlier?

```js
this.page = Number(m.route.param('page')) || 1;
```
or is that too soon?

Reply:
I'm not able to test this right now, but wouldn't scrolling up from page 2 back to page 1 keep page 2 in the querystring ?

Reply:
@clarkwinkelmann That is correct. There are no scroll handlers in the DiscussionList right now...

@luceos Hm, perhaps, haven't tested it.